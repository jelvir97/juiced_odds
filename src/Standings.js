import React, { useState, useEffect } from "react";
import NHL_API from "./NHL_API";
import StandingsTable from "./StandingsTable";
import { Tab } from "@headlessui/react";

const Standings = () => {
  const [standings, setStandings] = useState();

  useEffect(() => {
    const getStandings = async () => {
      const s = await NHL_API.getStandings();
      setStandings(s);
    };
    getStandings();
  }, []);


  const getClasses  = (selected)=>{
                return (
                    ' rounded-2xl px-4 '+ 
                  (selected
                    ? 'bg-black text-white shadow-lg font-bold '
                    : 'text-slate-600 hover:bg-blue-gray-100 hover:scale-110 duration-300')
                )
              }

  const makeTables = (s) => {
    return (
      
        <Tab.Group>
          <Tab.List className="w-full flex justify-between rounded-2xl border-2 border-slate-400 h-16 py-1 px-4 text-xl my-4">
            <Tab className={({selected})=>getClasses(selected)}>E - Metropolitan</Tab>
            <Tab className={({selected})=>getClasses(selected)}>E - Atlantic</Tab>
            <Tab className={({selected})=>getClasses(selected)}>W - Pacific</Tab>
            <Tab className={({selected})=>getClasses(selected)}>W - Central</Tab>
          </Tab.List>
          <Tab.Panels>

            <Tab.Panel>

                <StandingsTable division={s.Eastern.Metropolitan} name />
            </Tab.Panel>

            <Tab.Panel>

                <StandingsTable division={s.Eastern.Atlantic} name />
            </Tab.Panel>

            <Tab.Panel>

                <StandingsTable division={s.Western.Pacific} name />
            </Tab.Panel>

            <Tab.Panel>

                <StandingsTable division={s.Western.Central} name />
            </Tab.Panel>
          </Tab.Panels>
        </Tab.Group>

    );
  };

  return <div className="w-full px-2">{standings ? makeTables(standings) : ""}</div>;
};

export default Standings;
