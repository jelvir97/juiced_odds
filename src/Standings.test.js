import {render, screen, act, waitFor} from "@testing-library/react"
import Standings from "./Standings";
import userEvent from "@testing-library/user-event";

jest.mock('./NHL_API',() => ({
    ...jest.requireActual('./NHL_API'),
    getStandings: ()=> ({
        'Western': {
            'Pacific':[
                {
                    "gamesPlayed": 11,
                    "goalAgainst": 12,
                    "goalFor": 13,
                    "losses": 14,
                    "points": 15,
                    "streakCode": "W",
                    "streakCount": 1,
                    "teamName": {
                        "default": "test-team-1",
                    },
                    "teamAbbrev": {
                        "default": "tt1"
                    },
                    "ties": 16,
                    "wins": 17
                }
            ],
            'Central': [
                {
                    "gamesPlayed": 21,
                    "goalAgainst": 22,
                    "goalFor": 23,
                    "losses": 24,
                    "points": 25,
                    "streakCode": "W",
                    "streakCount": 2,
                    "teamName": {
                        "default": "test-team-2",
                    },
                    "teamAbbrev": {
                        "default": "tt2"
                    },
                    "ties": 26,
                    "wins": 27
                }
            ],
        },
        'Eastern':{
            'Atlantic':[
                {
                    "gamesPlayed": 31,
                    "goalAgainst": 32,
                    "goalFor": 33,
                    "losses": 34,
                    "points": 35,
                    "streakCode": "W",
                    "streakCount": 3,
                    "teamName": {
                        "default": "test-team-3",
                    },
                    "teamAbbrev": {
                        "default": "tt3"
                    },
                    "ties": 36,
                    "wins": 37
                }
            ],
            'Metropolitan':[
                {
                    "gamesPlayed": 41,
                    "goalAgainst": 42,
                    "goalFor": 43,
                    "losses": 44,
                    "points": 45,
                    "streakCode": "W",
                    "streakCount": 4,
                    "teamName": {
                        "default": "test-team-4",
                    },
                    "teamAbbrev": {
                        "default": "tt4"
                    },
                    "ties": 46,
                    "wins": 47
                }
            ]
        }     
    })
}));

test('renders', async()=>{
    await waitFor(()=>{
        render(
            <Standings />
        )
    })
})

test('matches snapshot', async()=>{
    await waitFor(()=>{
        render(
            <Standings />
        )
    })

    expect(screen.asFragment).toMatchSnapshot()
})

test('correct text', async()=>{
    await waitFor(()=>{
        render(
            <Standings />
        )
    })

    expect(await screen.findByText('test-team-4')).toBeInTheDocument()
})

test('tabs work', async()=>{
    const user = userEvent.setup()
    await waitFor(()=>{
        render(
            <Standings />
        )
    })

    expect(await screen.findByText('test-team-4')).toBeInTheDocument()

    await act(async()=>{
        await user.click(await screen.findByText('E - Atlantic'))
    })

    expect(await screen.findByText('test-team-3')).toBeInTheDocument()
    
})