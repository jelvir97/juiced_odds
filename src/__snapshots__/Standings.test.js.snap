// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correct text 1`] = `undefined`;

exports[`matches snapshot 1`] = `undefined`;
